clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7639,6 +7639,8 @@ def err_attribute_arm_mve_polymorphism : Error<
   "'__clang_arm_mve_strict_polymorphism' attribute can only be applied to an MVE/NEON vector type">;
 def err_attribute_webassembly_funcref : Error<
   "'__funcref' attribute can only be applied to a function pointer type">;
+def err_c23_underspecified_object_declaration: Error<
+  "'%select{struct|<ERROR>|union|<ERROR>|enum}0 %1' is defined as an underspecified object initializer">;
```

Comment:
I think that the diagnostic message need to be corrected.

Reply:
I don't think `underspecified object` is useful for users
Maybe `error: the deduced type of foo refers to a struct|union|enum bar which is defined in in initializer.`

Reply:
AFAIK `underspecified` is how the standard calls it. And `constexpr` declarations in C are also "underspecified".
gcc agrees https://godbolt.org/z/1KcW9zTcd .
I agree that `underspecified` sounds weird for ordinary people though.

Reply:
I also didn't wanted to just use the same wording as GCC.